### ISSUE TITLE
High CPU usage on logout when using remote Infinispan only setup

### ISSUE THREAD

**Issue by ahus1** [keycloak/keycloak-benchmark#933]
### Description

TL;DR: After the change of https://github.com/keycloak/keycloak/pull/31905 we have a query running on each logout. Now the nightly run shows the following behavior:

* an increased 50 perc latency on logout of 2+ second, but also other requests (possibly due to contention). 99 perc goes up to 8 seconds.
* very high CPU usage on Infinispan (from 15% to 850% percent CPU usage)
* There were 10_000 entries with a single owner in the session cache only (due to a misconfiguration in our setup, possibly originating from the persistent sessions re-configuration)
* Once the number of entries dropped to 0 again during the test, the CPU usage decreased.

(can be moved to the keycloak repository as needed) 

### Motivation

We need to have a fast logout. Before, the map existed in the user session. Not we need a fast query if we want to stay with a query. 

Further analysis is needed to see what Infinispan is doing. GC should be analysed on Infinispan.

These are a screenshots of a nightly run: The two quarter are not logging out, the second two quarters are logging out. During the third quarter, there are still client sessions in the cache. In the fourth quarter, they have expired. 

In the third quarter, Keycloak is slow.

![image](https://github.com/user-attachments/assets/13d28e75-7752-4fbf-a058-a1b7c01bfc85)

Number of user sessions goes down, but actually quite a lot later than the results return to normal response times:

![image](https://github.com/user-attachments/assets/5c13ede3-e739-4adb-9d7a-be38b8d64da4)

Number of client sessions don't expire on the same speed. This is strange it its own way. 

![image](https://github.com/user-attachments/assets/d90eeb43-1571-4ae1-84ba-68204b74eb13)
